### PR TITLE
[BEAM-1673] Fix PubSubIO write attribute issue

### PR DESCRIPTION
**Pull request by bchen-talend**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by jbonofre** [comment]
R: @jbonofre

**Comment by jbonofre** [comment]
retest this please

**Comment by dhalperi** [comment]
Hey @jbonofre  any comments here?

@tgroh maybe you can also take a look, since you've been paying attention to Pubsub recently?

File: sdks/java/core/src/test/java/org/apache/beam/sdk/util/PubsubJsonClientTest.java [hunk 1]
```diff
@@ -115,7 +118,7 @@ public void publishOneMessage() throws IOException {
         .encodeData(DATA.getBytes())
         .setAttributes(
             ImmutableMap.of(TIMESTAMP_LABEL, String.valueOf(MESSAGE_TIME),
```
Comment (tgroh) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/util/PubsubJsonClientTest.java]: This is increasingly difficult to read as you add more to it. Consider swapping to `ImmtuableMap.builder().put(...)...put(...).build()`

File: sdks/java/core/src/main/java/org/apache/beam/sdk/util/PubsubJsonClient.java [hunk 2]
```diff
@@ -135,7 +135,7 @@ public int publish(TopicPath topic, List<OutgoingMessage> outgoingMessages)
     for (OutgoingMessage outgoingMessage : outgoingMessages) {
       PubsubMessage pubsubMessage = new PubsubMessage().encodeData(outgoingMessage.elementBytes);
 
-      Map<String, String> attributes = pubsubMessage.getAttributes();
+      Map<String, String> attributes = outgoingMessage.attributes;
```
Comment (tgroh) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/util/PubsubJsonClient.java]: formerly, the `PubSub` message attributes would always be null, yes?

Is there any defined behavior about `timestampLabel` and `idLabel` not already being in the attributes? We'll overwrite them here if they're already specified.

**Comment by dhalperi** [comment]
R: @reuvenlax

**Comment by dhalperi** [comment]
My high-level thoughts: @bchen-talend this is a real bug you caught and thank you! The new test looks great!

Thomas gave you some suggestions for cleaning up the test spec, but I'm also happy to merge this as-is. Let me know what you'd like.

File: sdks/java/core/src/main/java/org/apache/beam/sdk/util/PubsubJsonClient.java (hunk 2 above)
Comment (bchen-talend) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/util/PubsubJsonClient.java]: yes,  message attributes always be null, except when using `timestampLabel` and `idLabel`.

**Comment by bchen-talend** [comment]
@dhalperi thanks, I have modified the test follow Thomas's suggestion.

**Comment by dhalperi** [comment]
Merge! Thanks @bchen-talend

**Pull request closed without merging**